### DISCUSSION TITLE
`vanilla-extract/integration` shouldn't depend on `vite`

### DISCUSSION THREAD

**Discussion by tony-scio** [vanilla-extract-css/vanilla-extract#1305] [General]
`vanilla-extract/integration` has a dependency on `vite`, but not all users use the `vite-plugin`. Vite is a heavy weight dependency and has pretty regular dependabot security issues (most recently https://nvd.nist.gov/vuln/detail/CVE-2024-23331). So it's unfortunate for webpack users to have to both install it and force resolve to newer versions when it's not used.

In cases like these, it's usually most appropriate for optional dependencies to be marked `peerDependencies` instead of `dependencies`. Is it possible for vite to be a peer dependency of `integration` and a hard dependency of `vite-plugin`?

**Comment by taozhou-glean** [comment]
+1, looks like `vite` is marked as peer dependency even on vite-plugin: https://github.com/vanilla-extract-css/vanilla-extract/pull/1240

**Comment by askoufis** [comment]
https://github.com/vanilla-extract-css/vanilla-extract/pull/1536 should address this.
